src/graphql/resolvers/GuideAlarm.ts

```diff
@@ -0,0 +1,35 @@
+import { GuideAlarm } from '@prisma/client';
+import { FieldNode } from 'graphql';
+import { prisma } from '../../prisma/db.js';
+import { Resolvers, WfsType } from '../gen/index.js';
+
+export const GuideAlarmResolver: Resolvers = {
+  Query: {
+    async guideAlarms(_parent, _args, _context, info) {
+      // Get the wfs types to query for from the graphql info object
+      const wfsTypesToQuery =
+        ((info.fieldNodes[0]?.selectionSet?.selections as FieldNode[])
+          .map((selection) => selection.name.value)
+          .filter((wfs) => wfs !== '__typename') as WfsType[]) ?? [];
+
+      const alarms = await prisma.guideAlarm.findMany({
+        where: {
+          wfs: {
+            in: wfsTypesToQuery,
+          },
+        },
+      });
+
+      // Convert the array of alarms to an object with the wfs as the key
+      return Object.fromEntries(alarms.map((alarm) => [alarm.wfs, alarm])) as Record<WfsType, GuideAlarm>;
```

Comment:
I have the confusion, I have read different opinions about having "variable" key objects against having arrays of objects with a specific key with value equals to the variable.
I like your approach and maybe you can enlighten me about this.

Reply:
Yeah, there's something to be said for either approach. In this case, I chose for it because there's a very limited number of values, and the values _should_ always exist. In an array, there's no certainty that a value will exist because you have to iterate over the array. This way, if it doesn't things will break early

Reply:
Anyway, we are lucky that GN and GS use the same wavefront sensors, then those names should not change, even after the desired wfs upgrade.